pkg/document/crdt/counter.go

```diff
@@ -59,32 +55,77 @@ type Counter struct {
 }
 
 // NewCounter creates a new instance of Counter.
-func NewCounter(value interface{}, createdAt *time.Ticket) *Counter {
-	switch val := value.(type) {
-	case int:
-		if math.MaxInt32 < val || math.MinInt32 > val {
+func NewCounter(valueType CounterType, value interface{}, createdAt *time.Ticket) *Counter {
+	switch valueType {
+	case IntegerCnt:
```

Comment:
Both `NewCounter` and `Increase` have the same logic that casts the given type to a predefined type. How about defining the below functions and removing duplicated logic from `NewCounter` and `Increase`?

`func castInt(value interface{}): int32`
`func castLong(value interface{}): int64`